athena-elasticsearch/src/test/java/com/amazonaws/connectors/athena/elasticsearch/integ/ElasticsearchIndexUtils.java

```diff
@@ -0,0 +1,89 @@
+/*-
+ * #%L
+ * athena-elasticsearch
+ * %%
+ * Copyright (C) 2019 - 2021 Amazon Web Services
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.amazonaws.connectors.athena.elasticsearch.integ;
+
+import com.amazonaws.connectors.athena.elasticsearch.AwsRestHighLevelClient;
+import com.amazonaws.connectors.athena.elasticsearch.AwsRestHighLevelClientFactory;
+import com.amazonaws.connectors.athena.elasticsearch.ElasticsearchDomainMapProvider;
+import org.elasticsearch.action.DocWriteRequest;
+import org.elasticsearch.action.index.IndexRequest;
+import org.elasticsearch.client.RequestOptions;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * Used to create Elasticsearch indices and insert documents for use with the Integration Tests.
+ */
+public class ElasticsearchIndexUtils implements AutoCloseable
+{
+    private static final Logger logger = LoggerFactory.getLogger(ElasticsearchIndexUtils.class);
+
+    private final AwsRestHighLevelClient client;
+    private final String index;
+
+    private Integer requestId;
+
+    public ElasticsearchIndexUtils(String domainName, String index) {
+        AwsRestHighLevelClientFactory clientFactory = new AwsRestHighLevelClientFactory(true);
+        ElasticsearchDomainMapProvider domainMapProvider = new ElasticsearchDomainMapProvider(true);
+        Optional<String> endpoint = Optional.ofNullable(domainMapProvider.getDomainMap(null).get(domainName));
```

Comment:
Won't passing in null here cause an exception?  

```
     * @throws RuntimeException The domain map cannot be created due to an error with the AWS ES SDK or an invalid
     * domainMapping variable (empty, null, or contain invalid information that cannot be parsed successfully).
     */
    public Map<String, String> getDomainMap(String domainMapping)
            throws RuntimeException
```

Reply:
No. This will only cause an exception when `autoDiscoverEndpoint` is **false** (see code below). In this case `autoDiscoverEndpoint` will always be **true**  because of the instantiation: `domainMapProvider` = new ElasticsearchDomainMapProvider(**true**);
```java
    protected Map<String, String> getDomainMap(String domainMapping)
            throws RuntimeException
    {
        if (autoDiscoverEndpoint) {
            // Get domain mapping via the AWS ES SDK (1.x).
            return getDomainMapFromAmazonElasticsearch();
        }
        else {
            // Get domain mapping from the domainMapping variable.
            return getDomainMapFromEnvironmentVar(domainMapping);
        }
    }
```